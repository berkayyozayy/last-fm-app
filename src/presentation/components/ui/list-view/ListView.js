import React from "react";
import styles from "./ListView.module.css";

function ListView({
  image,
  label,
  name,
  count,
  listeners,
  showListeners,
  url,
}) {
  return (
    <div className={styles["list-container"]}>
      <a href={url} target="_blank" rel="noreferrer">
        <div className={styles.list}>
          <div className={styles.card}>
            <div className={styles["image-wrapper"]}>
              <img src={image} alt={label} />
            </div>
            <div className={styles.content}>
              <p className={styles["name"]}>{name}</p>
              <span className={styles.title}>{label}</span>
              <hr />
              <span className={styles["play-count"]}>Playcount: {count}</span>
              {showListeners && (
                <span className={styles["listeners"]}>
                  Listeners: {listeners}
                </span>
              )}
            </div>
          </div>
        </div>
      </a>
      <hr />
    </div>
  );
}

export default ListView;
