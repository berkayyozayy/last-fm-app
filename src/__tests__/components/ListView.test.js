import { render } from "@testing-library/react";
import ListView from "../../presentation/components/ui/list-view/ListView";

describe("list view component", () => {
  test("it should render", () => {
    const { container } = render(<ListView />);
    expect(container).toMatchSnapshot();
  });
});
