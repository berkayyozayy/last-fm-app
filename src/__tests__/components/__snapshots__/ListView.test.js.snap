// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list view component it should render 1`] = `
<div>
  <div
    class="list-container"
  >
    <a
      rel="noreferrer"
      target="_blank"
    >
      <div
        class="list"
      >
        <div
          class="card"
        >
          <div
            class="image-wrapper"
          >
            <img />
          </div>
          <div
            class="content"
          >
            <p
              class="name"
            />
            <span
              class="title"
            />
            <hr />
            <span
              class="play-count"
            >
              Playcount: 
            </span>
          </div>
        </div>
      </div>
    </a>
    <hr />
  </div>
</div>
`;
